### PR TITLE
Update python dependencies

### PR DESCRIPTION
**Pull request by john-westcott-iv**
##### SUMMARY

General dependency update leaving most libraries pinned except where need for package updates. An unpinning will follow this after testing.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other

##### AWX VERSION

```
awx: 21.8.1.dev42+g147a48bf9f
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by djyasin — APPROVED** [review]

**Review by jay-steurer — APPROVED** [review]

File: requirements/requirements.in [hunk 1]
```diff
@@ -26,43 +25,41 @@ django-taggit
...
+twisted[tls]
 uWSGI
 uwsgitop
 wheel
 pip==21.2.4  # see UPGRADE BLOCKERs
-setuptools==58.2.0  # see UPGRADE BLOCKERs
-setuptools_scm[toml]>=3.4  # see UPGRADE BLOCKERs, xmlsec build dep
+setuptools  # see UPGRADE BLOCKERs
+setuptools_scm[toml]  # see UPGRADE BLOCKERs, xmlsec build dep
+xmlsec==1.3.12 # xmlsec 1.3.13 removed the ability to use lxml 4.7.0 but python3-saml requires lxml 4.7.0 so we need to pin xmlsec
```
Comment (AlanCoding) [inline requirements/requirements.in:62]: nice comments.

File: requirements/requirements.in [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
 ansiconv==1.0.0  # UPGRADE BLOCKER: from 2013, consider replacing instead of upgrading
 asciichartpy
 asn1
-autobahn>=20.12.3  # CVE-2020-35678
 azure-keyvault==1.1.0  # see UPGRADE BLOCKERs
 channels
-channels-redis>=3.1.0  # https://github.com/django/channels_redis/issues/212
-cryptography>=36.0.2,<37.0.0 # Until paramiko fixes https://github.com/paramiko/paramiko/issues/2038 we don't want to go to 37 or we end up with blowfish warnings in the job output
+channels-redis
+cryptography
```
Comment (AlanCoding) [inline requirements/requirements.in:8]: were the blowfish warnings resolved?

**Comment by jbradberry** [comment]
I think in the future, we should try to break these down into relatively isolated PRs.

- a single PR to catch all of the low-risk bumps of existing dependencies (i.e. no code changes, no expected interactions)
- a PR that catches the Python bootstraping packages bumps (i.e. higher risk, needs to be tested against multiple RHEL versions, probably has some sort of Productization implications)
- individual PRs for packages or small groups of interrelated packages that
  - involve code changes, or
  - potentially involve behavior changes, or
  - introduce a new dependency plus license file, or
  - involve an upgrade that is noted as a potential blocker

**Comment by john-westcott-iv** [comment]
> I think in the future, we should try to break these down into relatively isolated PRs.

That would be ideal but currently the updater script just blanket updates everything. If we wanted to break things down into smaller pieces it might be more manually intensive to do that and/or require changes to  the updater script.

**Comment by jbradberry** [comment]
> That would be ideal but currently the updater script just blanket updates everything. If we wanted to break things down into smaller pieces it might be more manually intensive to do that and/or require changes to the updater script.

If I remember right, only when you clear entries out of the requirements.txt file do they get bumped.  It ought to be possible to surgically remove specific lines from the requirements.txt to only update those.

**Review by akus062381 — APPROVED** [review]

File: requirements/requirements.in (hunk 2 above)
Comment (john-westcott-iv) [reply requirements/requirements.in:8]: We did manual testing with SSH keys and `connection: paramiko` and did not find any blowfish warnings.

**Pull request merged**